android-components/components/support/utils/src/main/java/mozilla/components/support/utils/DownloadUtils.kt

```diff
@@ -344,7 +344,7 @@ object DownloadUtils {
             // Compare the last segment of the extension against the mime type.
             // If there's a mismatch, discard the entire extension.
             val typeFromExt = mimeTypeMap.getMimeTypeFromExtension(filename.substringAfterLast('.'))
-            if (typeFromExt?.equals(mimeType, ignoreCase = true) != false) {
+            if (typeFromExt?.equals(mimeType, ignoreCase = true) != true) {
```

Comment:
@jackyzy823  the first check here is to make sure we found a corresponding `mimetype` before we compare the file extension. But now it check that we didn't find the corresponding mime type before comparing it.